### PR TITLE
[Numeral notations] Use Coqlib registered constants

### LINKED ISSUE #8564: Numeral notations should maybe use Coqlib for constants
Is this the wrong way to do things?  If so, it should be fixed in master:
https://github.com/coq/coq/blob/7cc70b0df61718a946327d5bfb056b140eeb54ba/plugins/syntax/numeral.ml#L36-L40

### PR DESCRIPTION
**Pull request by vbgl**
Fixes #8564.

Waiting for HoTT release, see https://github.com/coq/coq/pull/8720#issuecomment-434597150

### CODE REVIEW

File: theories/Init/Decimal.v
Comment (SkySkimmer) [inline theories/Init/Decimal.v]: Why this change?

Comment (vbgl) [reply theories/Init/Decimal.v]: I agree it’s unrelated. But since I was already touching this file, I took the opportunity to tidy it a bit…

File: plugins/syntax/numeral.ml
Comment (JasonGross) [inline plugins/syntax/numeral.ml]: What is the reason for dropping the parenthetical?  You could instead augment it to say `require BinNums or Register a constant as ... first`.

Either way, please also update the documentation to match the updated error message.

**Review by JasonGross — CHANGES_REQUESTED** [review]
Thanks!

**Comment by vbgl** [comment]
I’ve just updated the error messages and the corresponding documentation.

**Review by Zimmi48 — APPROVED** [review]

File: plugins/syntax/numeral.ml
Comment (JasonGross) [inline plugins/syntax/numeral.ml]: I think you should only display the part about `Require ` if the types don't resolve / aren't registered.  But I don't object strongly.

**Review by JasonGross — CHANGES_REQUESTED** [review]

**Comment by vbgl** [comment]
This PR breaks HoTT; the overlay is ready but would break the compatibility of HoTT with Coq 8.9. Thus we wait: 1. that Coq 8.9 is released (or at least the β); 2. that a version of HoTT compatible with said version of Coq is officially tagged; 3. that HoTT merges the overlay.

No need to hurry.

**Review by ppedrot — COMMENTED** [review]
I'll let people more knowledgeable comment.

**Comment by SkySkimmer** [comment]
>needs: merge of dependency

What's the dependency?

**Comment by SkySkimmer** [comment]
I've only seen that tag used for other Coq PRs before but I guess nothing else fits better.

**Comment by vbgl** [comment]
HoTT has been updated, so this PR is ready now.

**Review by JasonGross — APPROVED** [review]
My requests about error messages stand, but I'm not going to block this PR on that basis, and it would be nice to see this land.

**Comment by ejgallego** [comment]
Taking this as part of @coq/parsing-maintainers

**Review by ejgallego — APPROVED** [review]
Look pretty straightforward, will merge soon if not further comments.

**Pull request merged**
